### ISSUE TITLE
Unable to use SSH keys as machine credentials - Job hangs with very little output

### ISSUE THREAD

**Issue by Jimbo4794** [ansible/awx#11613]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Summary

Hi

I have been trying to setup a awx tower for our infrastucture and have hit a problem trying to get SSH keys working for machine credentials.  During setup I am using a very simple playbook to test communication:
```
---
- name: Hello World!
  hosts: all
 
  tasks:
 
  - name: Hello World!
    shell: echo "Hi! Tower is working."
 ```
 First tests of adding a host which would use a provided username/ssh-private-key combo credential and running the above playback resulted in k8s pod being brought up and entering a running state. The log from AWX browser page stay completely empty and the job hangs indefinatly. Looking at the pod logs there are two log lines:
 ```
 {
  "status": "starting",
  "runner_ident": "55",
  "command": [
    "ssh-agent",
    "sh",
    "-c",
    "trap 'rm -f /runner/artifacts/55/ssh_key_data' EXIT && ssh-add /runner/artifacts/55/ssh_key_data && rm -f /runner/artifacts/55/ssh_key_data && ansible-playbook -u james-test -vvvv -i /runner/inventory/hosts -e @/runner/env/extravars hello.yaml"
  ],
  "env": {
    "AWX_CICSK8S_SERVICE_PORT_80_TCP_PROTO": "tcp",
    "AWX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_PORT": "tcp://10.105.178.35:8443",
    "HOSTNAME": "automation-job-55-wfw45",
    "AWX_CICSK8S_SERVICE_PORT_80_TCP_ADDR": "10.96.184.204",
    "KUBERNETES_PORT_443_TCP_PROTO": "tcp",
    "KUBERNETES_PORT_443_TCP_ADDR": "10.96.0.1",
    "container": "oci",
    "AWX_CICSK8S_SERVICE_PORT_80_TCP_PORT": "80",
    "AWX_CICSK8S_SERVICE_SERVICE_PORT": "80",
    "AWX_CICSK8S_SERVICE_SERVICE_PORT_HTTP": "80",
    "KUBERNETES_PORT": "tcp://10.96.0.1:443",
    "PWD": "/runner",
    "HOME": "/home/runner",
    "AWX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_PORT_8443_TCP_PORT": "8443",
    "KUBERNETES_SERVICE_PORT_HTTPS": "443",
    "AWX_CICSK8S_SERVICE_PORT_80_TCP": "tcp://10.96.184.204:80",
    "KUBERNETES_PORT_443_TCP_PORT": "443",
    "AWX_CICSK8S_SERVICE_SERVICE_HOST": "10.96.184.204",
    "KUBERNETES_PORT_443_TCP": "tcp://10.96.0.1:443",
    "AWX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_PORT_8443_TCP": "tcp://10.105.178.35:8443",
    "AWX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_SERVICE_PORT_HTTPS": "8443",
    "AWX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_SERVICE_PORT": "8443",
    "AWX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_PORT_8443_TCP_PROTO": "tcp",
    "AWX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_SERVICE_HOST": "10.105.178.35",
    "SHLVL": "0",
    "KUBERNETES_SERVICE_PORT": "443",
    "AWX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_PORT_8443_TCP_ADDR": "10.105.178.35",
    "AWX_CICSK8S_SERVICE_PORT": "tcp://10.96.184.204:80",
    "PATH": "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
    "KUBERNETES_SERVICE_HOST": "10.96.0.1",
    "LC_CTYPE": "C.UTF-8",
    "ANSIBLE_FORCE_COLOR": "True",
    "ANSIBLE_HOST_KEY_CHECKING": "False",
    "ANSIBLE_INVENTORY_UNPARSED_FAILED": "True",
    "ANSIBLE_PARAMIKO_RECORD_HOST_KEYS": "False",
    "AWX_PRIVATE_DATA_DIR": "/tmp/awx_55_n8gknh_k",
    "JOB_ID": "55",
    "INVENTORY_ID": "3",
    "PROJECT_REVISION": "c67e332c59a526356f0ef452c127fc7ad6f58bd8",
    "ANSIBLE_RETRY_FILES_ENABLED": "False",
    "MAX_EVENT_RES": "700000",
    "AWX_HOST": "https://towerhost",
    "ANSIBLE_SSH_CONTROL_PATH_DIR": "/runner/cp",
    "ANSIBLE_COLLECTIONS_PATHS": "/runner/requirements_collections:~/.ansible/collections:/usr/share/ansible/collections",
    "ANSIBLE_ROLES_PATH": "/runner/requirements_roles:~/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles",
    "PYTHONPATH": ":/usr/local/lib/python3.8/site-packages/ansible_runner/config/../callbacks",
    "ANSIBLE_CALLBACK_PLUGINS": "/usr/local/lib/python3.8/site-packages/ansible_runner/config/../callbacks",
    "ANSIBLE_STDOUT_CALLBACK": "awx_display",
    "AWX_ISOLATED_DATA_DIR": "/runner/artifacts/55",
    "RUNNER_OMIT_EVENTS": "False",
    "RUNNER_ONLY_FAILED_EVENTS": "False"
  },
  "cwd": "/runner/project"
}
{
  "status": "running",
  "runner_ident": "55"
}
```
Which does not change no matter on the log level provided.
After some testing and swithcing to username/password combo credential the playbook and job run fine, so there are no connectivity issues. I continued to do more testing, letting the pod getting to its stuck running state, creating a remote shell onto it, running:
```
eval `ssh-agent`
ssh-add /path/to/private/key 
```
This would then allow me to run a ssh session from the pod to the machine in question. I had to do this by copying a private key onto the pod manually as the ssh-key-data found in /runner/artifacts/ID/ssh-key-data would leave the `ssh-add` command hanging.
Unsure on the process on which awx brings up pods to run the jobs in, and how the ssh data is replicated across to the pods im unsure on what to try next. This is looking to me like there is a issue with the ssh-key data that is being put on the pod. 

### AWX version

AWX 19.5.1

### Installation method

kubernetes

### Modifications

no

### Ansible version

ansible [core 2.11.8rc1.post0] - provided from quay.io/ansible/awx-ee:latest Execution env

### Operating system

Kubernetes - used 0.15.0 Operator to install

### Web browser

Firefox, Chrome, Safari, Edge

### Steps to reproduce

This setup has only just been setup using the 0.15.0 operator, then using the `make deploy` command to bring up the operator. A very simple AWX instance was setup:
```
apiVersion: awx.ansible.com/v1beta1
kind: AWX
metadata:
  name: awx
  namespace: awx
spec:
  service_type: clusterip
  node_selector: |
    kubernetes.io/arch: amd64
  postgres_selector: |
    kubernetes.io/arch: amd64
```
From this point, all I have done is setup a simple playbook, a single host, the credentials for the host, template and a project for running against the host. Nothing fancy has been setup. Credential types are set to machine, and a username and SSH Private Key provided

### Expected results

To access the provided host via ssh keys

### Actual results

Job hangs, unless username password credentials are provided

**Comment by alex-pimperton** [comment]
I wonder if this is related to this Ansible bug:

https://github.com/ansible/ansible/issues/76737

**Comment by TheOnion7** [comment]
I have exactly the same issue . I'm using AWX version 15.0.1. I then upgrade to AWX 19 and I still face the same issue. It seems to be an issue only for some hosts and not for others but I was not able to pinpoint why this is the case. Thank You.

**Comment by iv1111** [comment]
Hi, everyone. I'm using a standalone version of the ansible-runner and have the same problem.

`ssh-add /tmp/ansible-runner20220209-712921-1crfi4g/artifacts/result/ssh_key_data hangs indefinitely with no stdout and no stderr output.`

This is what ansible-runner does when one starts the playbook. Looks like ssh_key_data is a pipe and nothing gets read from it. On the other hand /tmp/ansible-runner20220209-703861-6gj34r/env/**ssh_key** is present and well-formed. I can open the contents.

**Comment by iv1111** [comment]
Similar issue: https://github.com/ansible/ansible-runner/issues/993

**Comment by TheOnion7** [comment]
Hi guys, 
I've fixed my problem. I hope it will be something similar for you . After we have looked for every possible causes, we found that Crowdstrike Falcon-Sensor pushed an update after Log4J vuln appeared and it disables the execution script which prevent the FIFO pipeline that ssh-key-data uses to get the credentials to AWX. 

We only had to disable falcon-sensor with **systemctl stop falcon-sensor.service** and try to execute again the template. It worked. 
For us, the solution was pretty basic but I didn't think of it right away. Hope it will help you too. 

You can take a look at this article : https://stackoverflow.com/questions/70320452/awx-all-jobs-stop-processing-and-hang-indefinitely-why

**Comment by nixocio** [comment]
We heard that Crowdstrike has fixed this issue on their end.

**Comment by auroraqin** [comment]
Hi @TheOnion7, @nixocio, I'm using AWX 19.5.1 on Openshift and met the same issue, do we have any other solution to fix the issue besides of disable falcon-sensor.service? Since I don't have permission to do that and we are not allowed to disable security.

**Comment by TheOnion7** [comment]
@auroraqin , unfortunately, from what I've seen , I never found any other way to fix the issue. It was an issue that I encountered with my last job, I changed job since that time. But I know that we opened a ticket with Crowdstrike about that issue. 

Very sorry about that but I think the best way would be to open a ticket with Crowsdstrike and see what they can do. @nixocio heard that Crowdstrike have fixed the issue. If you encountered this issue recently, did you make sure that the crowdstrike is up-to-date ? It might fix the issue, but to be honest, a few days before I quit my last job, I also saw the same problem. Might be related to crowdstrike version though. Let me know if I can do anything, take a quick look at your crowdstrike version just to make sure you're on the latest version. 

If you're on the latest one, I might be able to do some test on my side too and might be able to help you using local system. Let me know if I can do something.
